src/kvstore/NebulaStore.cpp

```diff
@@ -392,6 +392,9 @@ void NebulaStore::removeSpace(GraphSpaceID spaceId, bool isListener) {
   if (!isListener) {
     auto spaceIt = this->spaces_.find(spaceId);
     if (spaceIt != this->spaces_.end()) {
+      for (auto& [partId, part] : spaceIt->second->parts_) {
+        removePart(spaceId, partId, false);
```

Comment:
If added removePart here,  the `engine->removePart(partId);` will not needed below?

Reply:
actually need it too

Reply:
Why? Because there is also `e->removePart(partId);` in each `NebulaStore::removePart`, are they duplicated?

Reply:
engine remove parts by its own part's info, removing old codes in case of uncertainty is not a good idea.

Reply:
are you certain that this code could be removed? if you are certain I will remove it.

Reply:
I think it is, but need some tests. You could also sort out the logic to point out it, after all, we could not at code without considering the context.

Reply:
then who could do the tests? I am certain that this code could stay.